<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from svanM2_robot.xacro             | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="svanm2">
  <gazebo>
    <plugin filename="libmotor_controller.so" name="motor_controller">
      <!-- 
            <kp0>75</kp0>
            <kp1>30.2</kp1>
            <kp2>18.415</kp2> -->
      <kp0>200.2</kp0>
      <kp1>60.2</kp1>
      <kp2>42.415</kp2>
      <kd0>1</kd0>
      <kd1>1.5</kd1>
      <kd2>0.5</kd2>
      <!-- 
            <kp0>80</kp0>
            <kp1>40</kp1>
            <kp2>20</kp2>
            <kd0>1</kd0>
            <kd1>1</kd1>
            <kd2>0.5</kd2> -->
      <!-- 
            <kp0>200</kp0>
            <kp1>120</kp1>
            <kp2>60</kp2>
            <kd0>1</kd0>
            <kd1>1</kd1>
            <kd2>0.5</kd2> -->
      <!-- adding higher gains for the swing leg -->
      <kpsw0>300</kpsw0>
      <kpsw1>300</kpsw1>
      <kpsw2>300</kpsw2>
      <kdsw0>6</kdsw0>
      <kdsw1>6</kdsw1>
      <kdsw2>6</kdsw2>
    </plugin>
  </gazebo>
  <!-- IMU sensor -->
  <gazebo>
    <plugin filename="libgazebo_ros_imu.so" name="imu_sensor">
      <alwaysOn>true</alwaysOn>
      <bodyName>base</bodyName>
      <frameName>base</frameName>
      <topicName>/shvan/imu</topicName>
      <serviceName>imu_service</serviceName>
      <xyzOffset> 0 0 0 </xyzOffset>
      <rpyOffset> 0 0 0 </rpyOffset>
      <gaussianNoise>0.001</gaussianNoise>
      <updateRate>1000.0</updateRate>
    </plugin>
  </gazebo>
  <!-- joint and base sensor -->
  <gazebo>
    <plugin filename="libjoint_state_sensor.so" name="joint_state_sensor"/>
  </gazebo>
  <gazebo>
    <plugin filename="libbase_state_sensor.so" name="base_state_sensor"/>
  </gazebo>
  <!-- FR foot -->
  <gazebo reference="foot_FR">
    <material>Gazebo/Black</material>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0</restitution_coefficient>
          <threshold>1e+06</threshold>
        </bounce>
        <friction>
          <torsional>
            <coefficient>0.3</coefficient>
          </torsional>
          <ode>
            <mu>100.0</mu>
            <mu2>50</mu2>
          </ode>
          <bullet>
            <rolling_friction>0.3</rolling_friction>
            <friction2>50</friction2>
          </bullet>
        </friction>
        <contact>
          <bullet>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </bullet>
          <ode>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </ode>
        </contact>
      </surface>
    </collision>
  </gazebo>
  <!-- FL foot -->
  <gazebo reference="foot_FL">
    <material>Gazebo/Black</material>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0</restitution_coefficient>
          <threshold>1e+06</threshold>
        </bounce>
        <friction>
          <torsional>
            <coefficient>0.3</coefficient>
          </torsional>
          <ode>
            <mu>100.0</mu>
            <mu2>50</mu2>
          </ode>
          <bullet>
            <rolling_friction>0.3</rolling_friction>
            <friction2>50</friction2>
          </bullet>
        </friction>
        <contact>
          <bullet>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </bullet>
          <ode>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </ode>
        </contact>
      </surface>
    </collision>
  </gazebo>
  <!-- FL foot -->
  <gazebo reference="foot_RL">
    <material>Gazebo/Black</material>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0</restitution_coefficient>
          <threshold>1e+06</threshold>
        </bounce>
        <friction>
          <torsional>
            <coefficient>0.3</coefficient>
          </torsional>
          <ode>
            <mu>100.0</mu>
            <mu2>50</mu2>
          </ode>
          <bullet>
            <rolling_friction>0.3</rolling_friction>
            <friction2>50</friction2>
          </bullet>
        </friction>
        <contact>
          <bullet>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </bullet>
          <ode>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </ode>
        </contact>
      </surface>
    </collision>
  </gazebo>
  <!-- RR foot -->
  <gazebo reference="foot_RR">
    <material>Gazebo/Black</material>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0</restitution_coefficient>
          <threshold>1e+06</threshold>
        </bounce>
        <friction>
          <torsional>
            <coefficient>0.3</coefficient>
          </torsional>
          <ode>
            <mu>100.0</mu>
            <mu2>50</mu2>
          </ode>
          <bullet>
            <rolling_friction>0.3</rolling_friction>
            <friction2>50</friction2>
          </bullet>
        </friction>
        <contact>
          <bullet>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </bullet>
          <ode>
            <kp>30000.0</kp>
            <kd>1000.0</kd>
          </ode>
        </contact>
      </surface>
    </collision>
  </gazebo>
  <!-- Foot contacts. -->
  <!-- <gazebo>
          <plugin name="contact_plugin" filename="libcontact_plugin_new.so">
            <link1>foot_FR</link1>
            <link2>foot_FL</link2>
            <link3>foot_RL</link3>
            <link4>foot_RR</link4>
            <contact_model>0</contact_model>
          </plugin>
      </gazebo> -->
  <gazebo reference="foot_FR">
    <sensor name="FR_contact_sensor" type="contact">
      <always_on>1</always_on>
      <plugin filename="libcontact_plugin_default.so" name="contact_plugin_default">
        <foot_link>FR</foot_link>
      </plugin>
      <contact>
        <collision>foot_FR_collision</collision>
      </contact>
    </sensor>
  </gazebo>
  <gazebo reference="foot_FL">
    <sensor name="FL_contact_sensor" type="contact">
      <always_on>1</always_on>
      <plugin filename="libcontact_plugin_default.so" name="contact_plugin_default">
        <foot_link>FL</foot_link>
      </plugin>
      <contact>
        <collision>foot_FL_collision</collision>
      </contact>
    </sensor>
  </gazebo>
  <gazebo reference="foot_RL">
    <sensor name="RL_contact_sensor" type="contact">
      <always_on>1</always_on>
      <plugin filename="libcontact_plugin_default.so" name="contact_plugin_default">
        <foot_link>RL</foot_link>
      </plugin>
      <contact>
        <collision>foot_RL_collision</collision>
      </contact>
    </sensor>
  </gazebo>
  <gazebo reference="foot_RR">
    <sensor name="RR_contact_sensor" type="contact">
      <always_on>1</always_on>
      <plugin filename="libcontact_plugin_default.so" name="contact_plugin_default">
        <foot_link>RR</foot_link>
      </plugin>
      <contact>
        <collision>foot_RR_collision</collision>
      </contact>
    </sensor>
  </gazebo>
  <!-- <xacro:include filename="$(find m2_description)/xacro/transmission.xacro"/> -->
  <!-- <xacro:include filename="$(find go2_description)/xacro/materials.xacro"/> -->
  <!-- <xacro:include filename="$(find go2_description)/xacro/stairs.xacro"/> -->
  <link name="base_dummy">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="base_dummy_j" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_dummy"/>
    <child link="base"/>
  </joint>
  <link name="base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/BASE.STL" scale="1 1 1"/>
      </geometry>
      <material name="svan"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/BASE.STL"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0047249 -1.17e-05 0.0011585"/>
      <mass value="3.4234"/>
      <inertia ixx="0.0163964" ixy="-1.59e-05" ixz="0.0010427" iyy="0.0425922" iyz="-6e-07" izz="0.051134"/>
    </inertial>
  </link>
  <!-- abd_l-->
  <link name="FR_abd_l">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0492121 -0.0779589 0.0008006"/>
      <mass value="1.350949"/>
      <inertia ixx="0.0033376" ixy="-0.0001656" ixz="9e-06" iyy="0.0011263" iyz="-5.1e-06" izz="0.0033905"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_HIP.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_HIP.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- abd_j-->
  <joint name="FR_abd_j" type="revolute">
    <origin rpy="0 0 0" xyz="0.155 -0.06 0.0"/>
    <parent link="base"/>
    <child link="FR_abd_l"/>
    <axis xyz="1 0 0"/>
    <limit effort="20" lower="-4.8" upper="4.8" velocity="52.4"/>
  </joint>
  <!-- hip_l-->
  <link name="FR_hip_l">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.2e-06 -0.0008381 -0.0603416"/>
      <mass value="0.2502465"/>
      <inertia ixx="0.0011437" ixy="0" ixz="-1e-07" iyy="0.0011591" iyz="2.43e-05" izz="7.2e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_THIGH.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_THIGH.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- hip_j -->
  <joint name="FR_hip_j" type="revolute">
    <origin rpy="0 0 0" xyz="0.043751 -0.0796 0"/>
    <parent link="FR_abd_l"/>
    <child link="FR_hip_l"/>
    <axis xyz="0 -1 0"/>
    <limit effort="4.188790205" lower="55" upper="-4.047197551" velocity="28.6"/>
  </joint>
  <!-- knee_l-->
  <link name="FR_knee_l">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0035383 -2.73e-05 -0.068842"/>
      <mass value="0.20174"/>
      <inertia ixx="0.0010436" ixy="-1e-07" ixz="-6.93e-05" iyy="0.0010692" iyz="-1e-06" izz="4.24e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_SHANK.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_SHANK.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- knee_j-->
  <joint name="FR_knee_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1915"/>
    <parent link="FR_hip_l"/>
    <child link="FR_knee_l"/>
    <axis xyz="0 -1 0"/>
    <limit effort="4.916297857" lower="55" upper="-4.696533694" velocity="28.6"/>
  </joint>
  <!-- foot-->
  <link name="foot_FR">
    <contact>
      <friction_anchor/>
      <stiffness value="30000"/>
      <damping value="750"/>
      <spinning_friction value="0.3"/>
      <lateral_friction value="50"/>
    </contact>
    <inertial>
      <origin rpy="0 0 0" xyz="0.001497984407 2e-15 7.77e-05"/>
      <mass value="0.025"/>
      <inertia ixx="2e-06" ixy="0" ixz="0" iyy="2e-06" iyz="0" izz="2e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
  </link>
  <!-- dummy_j -->
  <joint name="FR_dummy_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.16693"/>
    <parent link="FR_knee_l"/>
    <child link="foot_FR"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="0" upper="0" velocity="0"/>
  </joint>
  <!-- abd_l-->
  <link name="FL_abd_l">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0492121 0.0779589 0.0008006"/>
      <mass value="1.350949"/>
      <inertia ixx="0.0033376" ixy="0.0001656" ixz="9e-06" iyy="0.0011263" iyz="5.1e-06" izz="0.0033905"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_HIP.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_HIP.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- abd_j-->
  <joint name="FL_abd_j" type="revolute">
    <origin rpy="0 0 0" xyz="0.155 0.06 0.0"/>
    <parent link="base"/>
    <child link="FL_abd_l"/>
    <axis xyz="1 0 0"/>
    <limit effort="20" lower="-4.8" upper="4.8" velocity="52.4"/>
  </joint>
  <!-- hip_l-->
  <link name="FL_hip_l">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.2e-06 0.0008381 -0.0603416"/>
      <mass value="0.2502465"/>
      <inertia ixx="0.0011437" ixy="0" ixz="-1e-07" iyy="0.0011591" iyz="-2.43e-05" izz="7.2e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_THIGH.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_THIGH.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- hip_j -->
  <joint name="FL_hip_j" type="revolute">
    <origin rpy="0 0 0" xyz="0.043751 0.0796 0"/>
    <parent link="FL_abd_l"/>
    <child link="FL_hip_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="4.188790205" lower="55" upper="-4.047197551" velocity="28.6"/>
  </joint>
  <!-- knee_l-->
  <link name="FL_knee_l">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0035383 2.73e-05 -0.068842"/>
      <mass value="0.20174"/>
      <inertia ixx="0.0010436" ixy="1e-07" ixz="-6.93e-05" iyy="0.0010692" iyz="1e-06" izz="4.24e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_SHANK.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_SHANK.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- knee_j-->
  <joint name="FL_knee_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1915"/>
    <parent link="FL_hip_l"/>
    <child link="FL_knee_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="4.916297857" lower="55" upper="-4.696533694" velocity="28.6"/>
  </joint>
  <!-- foot-->
  <link name="foot_FL">
    <contact>
      <friction_anchor/>
      <stiffness value="30000"/>
      <damping value="750"/>
      <spinning_friction value="0.3"/>
      <lateral_friction value="50"/>
    </contact>
    <inertial>
      <origin rpy="0 0 0" xyz="0.001497984407 -2e-15 7.77e-05"/>
      <mass value="0.025"/>
      <inertia ixx="2e-06" ixy="0" ixz="0" iyy="2e-06" iyz="0" izz="2e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
  </link>
  <!-- dummy_j -->
  <joint name="FL_dummy_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.16693"/>
    <parent link="FL_knee_l"/>
    <child link="foot_FL"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="0" upper="0" velocity="0"/>
  </joint>
  <!-- abd_l-->
  <link name="RR_abd_l">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0492121 -0.0779589 0.0008006"/>
      <mass value="1.350949"/>
      <inertia ixx="0.0033376" ixy="0.0001656" ixz="-9e-06" iyy="0.0011263" iyz="-5.1e-06" izz="0.0033905"/>
    </inertial>
    <visual>
      <origin rpy="3.14159265359 3.14159265359 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_HIP.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="3.14159265359 3.14159265359 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_HIP.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- abd_j-->
  <joint name="RR_abd_j" type="revolute">
    <origin rpy="0 0 0" xyz="-0.155 -0.06 0.0"/>
    <parent link="base"/>
    <child link="RR_abd_l"/>
    <axis xyz="-1 0 0"/>
    <limit effort="20" lower="-4.8" upper="4.8" velocity="52.4"/>
  </joint>
  <!-- hip_l-->
  <link name="RR_hip_l">
    <inertial>
      <origin rpy="0 0 0" xyz="4.2e-06 -0.0008381 -0.0603416"/>
      <mass value="0.2502465"/>
      <inertia ixx="0.0011437" ixy="0" ixz="1e-07" iyy="0.0011591" iyz="2.43e-05" izz="7.2e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_THIGH.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_THIGH.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- hip_j -->
  <joint name="RR_hip_j" type="revolute">
    <origin rpy="0 0 0" xyz="-0.043751 -0.0796 0"/>
    <parent link="RR_abd_l"/>
    <child link="RR_hip_l"/>
    <axis xyz="0 -1 0"/>
    <limit effort="4.188790205" lower="55" upper="-4.047197551" velocity="28.6"/>
  </joint>
  <!-- knee_l-->
  <link name="RR_knee_l">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0035383 -2.73e-05 -0.068842"/>
      <mass value="0.20174"/>
      <inertia ixx="0.0010436" ixy="-1e-07" ixz="-6.93e-05" iyy="0.0010692" iyz="-1e-06" izz="4.24e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_SHANK.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_SHANK.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- knee_j-->
  <joint name="RR_knee_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1915"/>
    <parent link="RR_hip_l"/>
    <child link="RR_knee_l"/>
    <axis xyz="0 -1 0"/>
    <limit effort="4.916297857" lower="55" upper="-4.696533694" velocity="28.6"/>
  </joint>
  <!-- foot-->
  <link name="foot_RR">
    <contact>
      <friction_anchor/>
      <stiffness value="30000"/>
      <damping value="750"/>
      <spinning_friction value="0.3"/>
      <lateral_friction value="50"/>
    </contact>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.001497984407 2e-15 7.77e-05"/>
      <mass value="0.025"/>
      <inertia ixx="2e-06" ixy="0" ixz="0" iyy="2e-06" iyz="0" izz="2e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
  </link>
  <!-- dummy_j -->
  <joint name="RR_dummy_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.16693"/>
    <parent link="RR_knee_l"/>
    <child link="foot_RR"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="0" upper="0" velocity="0"/>
  </joint>
  <!-- abd_l-->
  <link name="RL_abd_l">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0492121 0.0779589 0.0008006"/>
      <mass value="1.350949"/>
      <inertia ixx="0.0033376" ixy="-0.0001656" ixz="-9e-06" iyy="0.0011263" iyz="5.1e-06" izz="0.0033905"/>
    </inertial>
    <visual>
      <origin rpy="3.14159265359 3.14159265359 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_HIP.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="3.14159265359 3.14159265359 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/RF_HIP.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- abd_j-->
  <joint name="RL_abd_j" type="revolute">
    <origin rpy="0 0 0" xyz="-0.155 0.06 0.0"/>
    <parent link="base"/>
    <child link="RL_abd_l"/>
    <axis xyz="-1 0 0"/>
    <limit effort="20" lower="-4.8" upper="4.8" velocity="52.4"/>
  </joint>
  <!-- hip_l-->
  <link name="RL_hip_l">
    <inertial>
      <origin rpy="0 0 0" xyz="4.2e-06 0.0008381 -0.0603416"/>
      <mass value="0.2502465"/>
      <inertia ixx="0.0011437" ixy="0" ixz="1e-07" iyy="0.0011591" iyz="-2.43e-05" izz="7.2e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_THIGH.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_THIGH.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- hip_j -->
  <joint name="RL_hip_j" type="revolute">
    <origin rpy="0 0 0" xyz="-0.043751 0.0796 0"/>
    <parent link="RL_abd_l"/>
    <child link="RL_hip_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="4.188790205" lower="55" upper="-4.047197551" velocity="28.6"/>
  </joint>
  <!-- knee_l-->
  <link name="RL_knee_l">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0035383 2.73e-05 -0.068842"/>
      <mass value="0.20174"/>
      <inertia ixx="0.0010436" ixy="1e-07" ixz="-6.93e-05" iyy="0.0010692" iyz="1e-06" izz="4.24e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_SHANK.STL"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://svanM2_description/meshes/LF_SHANK.STL"/>
      </geometry>
    </collision>
  </link>
  <!-- knee_j-->
  <joint name="RL_knee_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1915"/>
    <parent link="RL_hip_l"/>
    <child link="RL_knee_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="4.916297857" lower="55" upper="-4.696533694" velocity="28.6"/>
  </joint>
  <!-- foot-->
  <link name="foot_RL">
    <contact>
      <friction_anchor/>
      <stiffness value="30000"/>
      <damping value="750"/>
      <spinning_friction value="0.3"/>
      <lateral_friction value="50"/>
    </contact>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.001497984407 -2e-15 7.77e-05"/>
      <mass value="0.025"/>
      <inertia ixx="2e-06" ixy="0" ixz="0" iyy="2e-06" iyz="0" izz="2e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
  </link>
  <!-- dummy_j -->
  <joint name="RL_dummy_j" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.16693"/>
    <parent link="RL_knee_l"/>
    <child link="foot_RL"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="0" upper="0" velocity="0"/>
  </joint>
</robot>
